src/rpc/blockchain.cpp

```diff
@@ -2260,10 +2281,33 @@ UniValue dumptxoutset(const JSONRPCRequest& request)
                 }
         },
         RPCExamples{
-            HelpExampleCli("dumptxoutset", "utxo.dat")
+            HelpExampleCli("dumptxoutset", "utxo.dat") +
+            HelpExampleCli("dumptxoutset", "utxo.dat '[]'") +
+            HelpExampleCli("dumptxoutset", "utxo.dat '[\"txid\", \"vout\"]' false ':'")
         }
     }.Check(request);
 
+    // handle optional ASCII parameters
+    const bool is_compact = request.params[1].isNull();
+    const bool show_header = request.params[2].isNull() || request.params[2].get_bool();
+    const std::string separator = request.params[3].isNull() ? "," : request.params[3].get_str();
+    std::vector<std::pair<std::string, cb_t>> requested;
+    if (!is_compact) {
+        const auto& arr = request.params[1].get_array();
+        const std::unordered_map<std::string, cb_t> ascii_map(std::begin(ascii_types), std::end(ascii_types));
+        for(auto i = 0; i < arr.size(); ++i) {
```

Comment:
`auto` doesn't really work in this context...

```
warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
```

Reply:
Ha, funny that a compiler can get the warning but don't properly choose the type of `i`. I guess the `0` must confuse it. 

Thanks for catching this, I've just updated the branch,